UN-OCHA/unocha-site
# [UNO-514][UNO-605] ReliefWeb integration

Refs: UNO-514, UNO-605

This PR changes the white labeled reliefweb documents URLs to use `publications` as base path to simplify the logic and uses the `redirects` field in the API for the lookups to handle changes to the path aliases on RW.

It also provides a ReliefWeb River field, widget, formatter, templates and components used to display a list of RW documents from a RW river URL (ex: /updates). The documents links are white labeled by default (can be turned off) in the formatter.

This field type can be added to a paragraph, for example to build the "Latest updates" block with multiple rivers.

There is probably some refinements to do but it could possible also be used for the `publications` page by adding settings to the field type to show some filters and the number of results but that's for another PR.

Also the "view all" link that can be enabled via a formatter option currently doesn't change the URL so it points to the ReliefWeb river URL entered in the field. If we want to point at the `/publications` on unocha.org then we may need to adjust the field type or override the value in a preprocess hook.

<img width="1075" alt="Screenshot 2023-04-14 at 10 37 53" src="https://user-images.githubusercontent.com/696348/231919323-b60eb8bc-34be-4121-9cb7-9f3d15feed29.png">

## Notes

- The `redirects` field is not yet deployed in the RW API so the lookup using that field can be disabled in the meantime with `$config['unocha_reliefweb.settings']['reliefweb_api_use_redirects'] = FALSE;`
- Currently there is a hard coded filter to restrict to OCHA documents (RW documents with the `ocha_product` field). If we want to display other documents, then this could become a switch in the field settings.
- The templates, css etc. were copied form RW9 codebase. It's missing the icons for the "format" that come from the `rw-icons` component in RW. Not sure if we want to port this component as well. @left23 What do you think?

## Tests

Set the following in your `settings.local.php`:

```
$config['unocha_reliefweb.settings']['reliefweb_api_url'] = 'https://api.reliefweb.int/v1';
$config['unocha_reliefweb.settings']['reliefweb_website'] = 'https://reliefweb.int';
$config['unocha_reliefweb.settings']['reliefweb_api_use_redirects'] = FALSE;
```

1. Add a `ReliefWeb River` field to a content type or paragraph type.
2. Edit a node,  (add a paragraph), fill in the field with a RW river URL containing OCHA docs.
3. Save and check that the river appears.
